compiler/rustc_target/src/abi/mod.rs

```diff
@@ -238,22 +238,38 @@ pub enum Endian {
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Encodable, Decodable)]
 #[derive(HashStable_Generic)]
 pub struct Size {
+    // The top 3 bits are ALWAYS zero.
     raw: u64,
```

Comment:
I am concerned by this comment that is being added here.
- It doesn't seem actually enforced -- with `Size::from_bytes` I can easily construct a `Size` that violates this.
- I would like to use this type to represent the size of any possible Rust object in Miri (including for things like `size_of_val_raw`, where the object does not actually have to exist in the address space). However, the size limit for that is `isize::MAX`. Something seems off if the compiler's `Size` type cannot be used to represent the size of all objects in the language...

Currently Miri enforces `dl.obj_size_bound()` as the size limit, but that is probably wrong -- it does not match what the [reference](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) and [`size_of_val_raw`](https://doc.rust-lang.org/std/mem/fn.size_of_val_raw.html) say the limit should be (they both say it is `isize::MAX`).